### PR TITLE
feat(misconf): Support custom data for rego policies for cloud

### PR DESCRIPTION
**Pull request by simar7**
## Description

This PR adds support for supplying custom data for rego policies.

## Related issues
- Close https://github.com/aquasecurity/trivy/issues/4565

## Checklist
- [x] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [x] I've added tests that prove my fix is effective or that my feature works.
- [x] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

**Comment by knqyf263** [comment]
Hi Simar, I've removed this PR from the milestone as we usually have only issues in the milestone.

File: go.mod [hunk 1]
```diff
@@ -13,7 +13,7 @@ require (
 	github.com/NYTimes/gziphandler v1.1.1
 	github.com/alicebob/miniredis/v2 v2.30.3
 	github.com/aquasecurity/bolt-fixtures v0.0.0-20200903104109-d34e7f983986
-	github.com/aquasecurity/defsec v0.90.1
+	github.com/aquasecurity/defsec v0.90.2-0.20230705181508-367b32a91f64
```
Comment (simar7) [inline go.mod:16]: note: Update this to next defsec release once it is available.

File: pkg/cloud/aws/commands/run_test.go [hunk 2]
```diff
@@ -969,6 +971,7 @@ func Test_Run(t *testing.T) {
 		cacheContent string
 		regoPolicy   string
 		allServices  []string
+		inputData    string
```
Comment (AnaisUrlichs) [inline pkg/cloud/aws/commands/run_test.go:974]: Will we provide an example in the docs on using this?

**Review by AnaisUrlichs — APPROVED** [review]

File: pkg/cloud/aws/commands/run_test.go (hunk 2 above)
Comment (simar7) [reply pkg/cloud/aws/commands/run_test.go:986]: We already have that here https://aquasecurity.github.io/trivy/v0.43/docs/scanner/misconfiguration/custom/data/

This PR just extends that functionality to be used in Trivy Cloud Scanning.

File: pkg/cloud/aws/commands/run_test.go [hunk 3]
```diff
@@ -958,8 +971,9 @@ const expectedS3AndCloudTrailResult = `{
 `
 
 func Test_Run(t *testing.T) {
-
-	regoDir := t.TempDir()
+	regoDir := filepath.Join("testdata", "Test_Run_Dir")
+	require.NoError(t, os.MkdirAll(regoDir, 0755))
+	defer os.RemoveAll(regoDir)
```
Comment (knqyf263) [inline pkg/cloud/aws/commands/run_test.go:976]: I'm curious about the benefit of this change.

File: pkg/commands/artifact/run.go [hunk 4]
```diff
@@ -579,7 +579,7 @@ func initScannerConfig(opts flag.Options, cacheClient cache.Cache) (ScannerConfi
 			Trace:                   opts.Trace,
 			Namespaces:              append(opts.PolicyNamespaces, defaultPolicyNamespaces...),
 			PolicyPaths:             append(opts.PolicyPaths, downloadedPolicyPaths...),
-			DataPaths:               opts.DataPaths,
+			DataPaths:               append(opts.DataPaths, downloadedPolicyPaths...),
```
Comment (knqyf263) [inline pkg/commands/artifact/run.go:582]: Why do we want to pass policy paths as data paths? The policies are loaded as data by mistake, no? I may be missing something.

File: pkg/cloud/aws/commands/run_test.go (hunk 3 above)
Comment (simar7) [reply pkg/cloud/aws/commands/run_test.go:976]: https://github.com/open-policy-agent/opa/issues/4521

File: pkg/commands/artifact/run.go (hunk 4 above)
Comment (simar7) [reply pkg/commands/artifact/run.go]: it was a typo, updated here https://github.com/aquasecurity/trivy/pull/4745/commits/18b29d5d6e399de2a46fd00ebbb61a34fd01e32d

**Review by knqyf263 — APPROVED** [review]

**Pull request merged**